lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -1772,6 +1806,20 @@ static void populateTypeConversion(TypeConverter &typeConverter) {
     return IntegerType::get(type.getContext(), type.getWidth());
   });
 
+  typeConverter.addConversion([&](RealType type) {
+    MLIRContext *ctx = type.getContext();
+    mlir::FloatType fTy;
+    switch (type.getWidth()) {
+    case moore::RealWidth::f32:
+      fTy = mlir::Float32Type::get(ctx);
+      break;
+    case moore::RealWidth::f64:
+      fTy = mlir::Float64Type::get(ctx);
+      break;
+    }
+    return fTy;
```

Comment:
I think you can return the types directly, and if nothing matches in the switch statement, returning a null type will indicate conversion failure to the conversion framework:

```suggestion
    switch (type.getWidth()) {
    case moore::RealWidth::f32:
      return mlir::Float32Type::get(ctx);
    case moore::RealWidth::f64:
      return mlir::Float64Type::get(ctx);
    default:
      return mlir::Type{};
    }
```

Reply:
If we add a `-> mlir::Type` to the lambda signature this works. Done!